src/Neo/SmartContract/Native/LedgerContract.cs

```diff
@@ -47,13 +47,21 @@ internal override ContractTask OnPersist(ApplicationEngine engine)
             engine.Snapshot.Add(CreateStorageKey(Prefix_Block).Add(engine.PersistingBlock.Hash), new StorageItem(Trim(engine.PersistingBlock).ToArray()));
             foreach (TransactionState tx in transactions)
             {
+                // Remove possible previously saved malicious conflict records for the transaction (if any).
+                foreach (var (key, _) in engine.Snapshot.Find(CreateStorageKey(Prefix_Transaction).Add(tx.Transaction.Hash).ToArray()))
+                    engine.Snapshot.Delete(key);
```

Comment:
The discussion started by @roman-khimov and ported from https://github.com/AnnaShaleva/neo/commit/deb1470a6286d01b97c842d79c55117a101adef0#r127677374:
>Not sure about this. It's garbage collection effectively. Can slow down tx processing.

Reply:
I'd say that's a job for `RemoveUntraceableBlocks` option (C# node doesn't have it, but it can be implemented). This storage space is paid for at the same time with the new setting, so keeping these entries is OK.

Reply:
I think that we should have a maximum allowed signers too

Reply:
Having maximum allowed signers will bring us to the same problem that was described in https://github.com/neo-project/neo/issues/2907#issuecomment-1722506014. So any kind of such restriction leads to this possible attack when malicious conflicts prevent some real (and valid) conflict from entering the chain.

That's why we build this new solution in such way so that all conflicts data are properly paid. And it's even harder to spam the DB with Conflicts-related data than via contract storage.

Reply:
it's not necessary to care about those "dust" records and they don't really hurt.

correct attr fee introduced by @shargon is able to avoid the abuse and restrict the storage cost.

an uncontrollable O(n) procedure here can be avoided